src/systems/pincer.rs

```diff
@@ -0,0 +1,58 @@
+use crate::components::PincerAi::Attacking;
+use amethyst::{
+    core::Transform,
+    ecs::{Join, ReadStorage, System, WriteStorage},
+    renderer::{palette::Srgba, resources::Tint},
+};
+
+use crate::components::{Direction, Motion, Pincer, PincerAi};
+
+pub struct PincerAiSystem;
+
+/// Execute Pincer AI logic.
+/// Attack logic is exceedingly simple. Check the direction of the target and change the sign of
+/// the pincer's velocity to make it run towards the target.
+/// Also changes the pincers color tint depending on whether it's aggro or not.
+impl<'s> System<'s> for PincerAiSystem {
+    type SystemData = (
+        WriteStorage<'s, Pincer>,
+        WriteStorage<'s, Direction>,
+        WriteStorage<'s, Motion>,
+        WriteStorage<'s, Tint>,
+        ReadStorage<'s, Transform>,
+    );
+
+    fn run(
+        &mut self,
+        (mut pincers, mut directions, mut motions, mut tints, transforms): Self::SystemData,
+    ) {
+        for (pincer, direction, motion, tint, transform) in (
+            &mut pincers,
+            &mut directions,
+            &mut motions,
+            &mut tints,
+            &transforms,
+        )
+            .join()
+        {
+            if let Attacking { target } = pincer.ai {
+                if let Some(target_transform) = transforms.get(target) {
+                    let delta_distance =
+                        target_transform.translation().x - transform.translation().x;
+                    motion.velocity.x = delta_distance.signum() * motion.velocity.x.abs();
+                    direction.set_x_velocity(motion.velocity.x);
+                } else {
+                    // The target no longer exists, go back to patrolling.
+                    pincer.ai = PincerAi::Patrolling;
```

Comment:
There is a bug that I noticed when the marine dies. The pincer gets stuck at its current position and seems like is not able to decide which direction to go. To reproduce the issue just shoot the pincer and let it collide with the marine when it attacks.

This might be due to the fact that the marine entity is not being deleted when it dies (This is one of the many pending things to be done). But I am not sure. There might be some other reason as well.

Reply:
Yes, I am aware of that behaviour. I intentionally left it as is, because the dying logic for the marine hasn't been built yet. If the marine entity is removed, that should automatically clear the target. However, if there's a delay between the marine dying and the entity being removed, then we need additional code to clear the target.

Reply:
I have logged an issue for deleting the marine entity on dying https://github.com/amethyst/space-menace/issues/39 .